integration-test/background/click-to-load-youtube.js

```diff
@@ -0,0 +1,335 @@
+        } else {
+            continue
+        }
+
+        stats.total += 1
+
+        if (request.status === 'blocked') {
+            stats.blocked += 1
+        } else {
+            // Consider anything not blocked as allowed, so that block tests
+            // don't miss redirected requests or failed requests.
+            stats.allowed += 1
+        }
+    }
+
+    return { youTubeIframeApi, youTubeStandard, youTubeNocookie }
+}
+
+describe('Test YouTube Click To Load', () => {
+    beforeAll(async () => {
+        ({ browser, bgPage, teardown } = await harness.setup())
+
+        // Wait for the extension to load its configuration files.
+        await bgPage.waitForFunction(
+            () => (
+                window.dbg &&
+                window.dbg.tds &&
+                !window.dbg.tds.isInstalling &&
+                window.dbg.tds.ClickToLoadConfig &&
+                window.dbg.tds.tds.trackers &&
+                window.dbg.tds.tds &&
+                window.dbg.tds.tds.domains &&
+                window.dbg.tds.tds.domains['youtube.com'] === 'Google LLC'
+            ),
+            { polling: 10 }
+        )
+
+        // Overwrite the parts of the configuration needed for our tests.
+        await loadTestConfig(bgPage, testConfig)
+    })
+
+    afterAll(async () => {
+        // Restore the original configuration.
+        await unloadTestConfig(bgPage, testConfig)
+
+        try {
+            await teardown()
+        } catch (e) {}
+    })
+
+    it('CTL: YouTube request blocking/redirecting', async () => {
+        // Open the test page and start logging network requests.
+        const page = await browser.newPage()
+        const pageRequests = []
+        logPageRequests(page, pageRequests)
+
+        // Initially there should be a bunch of requests. The iframe_api should
+        // be redirected to our surrogate but otherwise YouTube requests should
+        // be blocked.
+        await page.goto(testSite, { waitUntil: 'networkidle2' })
+        {
+            const {
+                youTubeIframeApi, youTubeStandard, youTubeNocookie
+            } = summariseYouTubeRequests(pageRequests)
+
+            expect(youTubeIframeApi.checked).toBeTrue()
+            expect(youTubeIframeApi.alwaysRedirected).toBeTrue()
+            expect(youTubeStandard.total).toBeGreaterThan(5)
+            expect(youTubeStandard.blocked).toEqual(youTubeStandard.total)
+            expect(youTubeStandard.allowed).toEqual(0)
+            expect(youTubeNocookie.blocked).toEqual(youTubeNocookie.total)
+            expect(youTubeNocookie.allowed).toEqual(0)
+        }
+
+        // Once the user clicks to load a video, the iframe_api should be loaded
+        // and the video should be unblocked.
+        pageRequests.length = 0
+        const button = await page.evaluateHandle(
+            'document.querySelector("div:nth-child(2) > div")' +
+            '.shadowRoot.querySelector("button")'
+        )
+        await button.click()
```

Comment:
for facebook we are clicking all buttons, here we are clicking one, what's the logic behind this?